documentation/correctness-previous.adoc

```diff
@@ -0,0 +1,130 @@
+The reporting is typically dependent on the precise JSON parser used.
+In particular, the JSON parser used might recover automatically, essentially ignoring syntax errors without reporting them in any way.
+This could be problematic, e.g. in case of duplicate keys in an object: the JSON parser is most likely to only reproduce the last value for the duplicate key, ignoring the previous ones.
+
+=== Structural syntactic
+
+The JSON conforms to https://github.com/LionWeb-io/specification/blob/main/serialization/serialization.adoc[the specification for the JSON serialization format].
+Note that JSON Schema isn't powerful enough to express more high-level, nor constraints of a referential nature on the serialization chunk.
+
+Incorrectness is to be reported in terms of e.g. a https://github.com/json-path/JsonPath[JsonPath] to the location(s) of failure(s).
+This is typically dependent on the precise JSON Schema validator framework/library/utility used.
+Recovering from an incorrectness is sometimes possible, e.g.:
+
+. If extra properties are present, they can simply be removed or ignored.
+. If properties are missing, they could just be added, provided they have a compliant default value.
+
+=== Hierarchical
+
+The JSON doesn't violate the following constraints.
+
+. IDs of serialized nodes are unique (within the serialization).
+Violation is to be reported in terms of precise locations in the JSON file of all the serialized nodes with equal ID.
+. The `parent` property of a serialized node matches the `id` of the containing node (whenever that containing node is present in the serialization chunk).
+Concretely: if a node `N` is a child of a node `P` then `N.parent` should equal `P`.
+Violation is to be reported in terms of the ID of the contain**ing** serialized node, the ID of the contain**ed** serialized node, and the `parent` ID stated by that.
+. No duplicate containments, i.e. nodes are not the children of multiple other nodes, nor of themselves.
+Violation is to be reported in terms of the ID of the duplicately-contained serialized node, and the IDs of all serialized nodes that claim to contain it, including the links by which this happens, as well the index in the value of those links.
+. No containment cycles, i.e.: nodes don't contain themselves.
+. **TODO**  extend for annotations
+
+At this point, links (so: containments and references) don't need to resolve (yet).
+Concretely, this means that IDs that are members of the `children` (for containments) or `targets` (for references) arrays, don't need to resolve to a node in the same JSON serialization chunk.
+They could very well reside in another serialization.
+That pertains specifically to references to language entities of the LionCore builtins language, which is assumed to be imported/available implicitly.
+If all links do resolve, we say the serialization chunk is _self-contained_.
+
+=== Meta-referential
```

Comment:
If we report category of errors to the users (e.g., syntactic, semantic, lexical, etc.) and if these titles will end up being one of the categories, then I would look for a simpler name, so that a casual user of LionWeb could more easily figure out the meaning